src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -0,0 +1,153 @@
+import java.util.List;
+
+/** A subclass of ModelToViewTransformer which translates model into API tests in Java. */
+public class JavaGapicSurfaceTestTransformer implements ModelToViewTransformer {
+
+  static String TEST_TEMPLATE_FILE = "java/test.snip";
+
+  private GapicCodePathMapper pathMapper;
+
+  public JavaGapicSurfaceTestTransformer(GapicCodePathMapper javaPathMapper) {
+    this.pathMapper = javaPathMapper;
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> views = new ArrayList<>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(
+              service, apiConfig, createTypeTable(), new JavaSurfaceNamer());
+      addImports(context);
+      GapicSurfaceTestClassView testClass = createTestClassView(context);
+      views.add(testClass);
+    }
+    return views;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    List<String> fileNames = new ArrayList<>();
+    fileNames.add(TEST_TEMPLATE_FILE);
+    // TODO(shinfan): Add gRPC service mock template.
+    return fileNames;
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new JavaTypeTable(), new JavaModelTypeNameConverter());
+  }
+
+  private void addImports(SurfaceTransformerContext context) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    typeTable.saveNicknameFor("org.junit.After");
+    typeTable.saveNicknameFor("org.junit.AfterClass");
+    typeTable.saveNicknameFor("org.junit.Before");
+    typeTable.saveNicknameFor("org.junit.BeforeClass");
+    typeTable.saveNicknameFor("org.junit.Test");
+    typeTable.saveNicknameFor("java.io.IOException");
+    typeTable.saveNicknameFor("java.util.List");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.MockServiceHelper");
+    typeTable.saveNicknameFor("com.google.api.gax.testing.MockGrpcService");
+    typeTable.saveNicknameFor("com.google.protobuf.GeneratedMessage");
+    typeTable.saveNicknameFor("junit.framework.Assert");
+  }
+
+  private GapicSurfaceTestClassView createTestClassView(SurfaceTransformerContext context) {
+    Interface service = context.getInterface();
+    String outputPath = pathMapper.getOutputPath(service, context.getApiConfig());
+
+    GapicSurfaceTestClassView testClass =
+        GapicSurfaceTestClassView.newBuilder()
+            .packageName(context.getApiConfig().getPackageName())
+            .apiSettingsClassName(context.getNamer().getApiSettingsClassName(service))
+            .apiClassName(context.getNamer().getApiWrapperClassName(service))
+            .name(context.getNamer().getTestClassName(service))
+            .mockServiceClassName(context.getNamer().getMockServiceClassName(service))
+            .testCases(createTestCaseViews(context))
+            .outputPath(outputPath)
+            .imports(context.getTypeTable().getImports())
```

Comment:
Add a comment noting that the imports have to be after the other generation code is run because the type table is modified (see the comments to that effect in JavaGapicSurfaceTransformer)